roomserver/internal/query/query.go

```diff
@@ -830,13 +846,20 @@ func (r *Queryer) QueryCurrentState(ctx context.Context, req *api.QueryCurrentSt
 	return nil
 }
 
-func (r *Queryer) QueryRoomsForUser(ctx context.Context, req *api.QueryRoomsForUserRequest, res *api.QueryRoomsForUserResponse) error {
-	roomIDs, err := r.DB.GetRoomsByMembership(ctx, req.UserID, req.WantMembership)
+func (r *Queryer) QueryRoomsForUser(ctx context.Context, userID spec.UserID, desiredMembership string) ([]spec.RoomID, error) {
+	roomIDStrs, err := r.DB.GetRoomsByMembership(ctx, userID, desiredMembership)
 	if err != nil {
-		return err
+		return nil, err
 	}
-	res.RoomIDs = roomIDs
-	return nil
+	roomIDs := make([]spec.RoomID, len(roomIDStrs))
+	for i, roomIDStr := range roomIDStrs {
+		roomID, err := spec.NewRoomID(roomIDStr)
+		if err != nil {
+			return nil, nil
+		}
```

Comment:
This logic feels wrong. I think we should be either skipping over this roomID & returning all valid roomIDs, or returning an error here.
Returning no rooms and no error doesn't seem right.
I think simply skipping over the invalid roomID would be best.

Reply:
Also true! I meant to return the error here - also would invalid room IDs be common place enough for us to want to skip them if they came up? Only because I could maybe see this causing obscure errors in the future if we silently ignore them (although I guess we could warn/error the log as well).

Reply:
Unsure. I don't really have a strong preference either way. If you think returning the error is the way to go then let's just do that.